src/NLog.StructuredLogging.Json/Helpers/ObjectExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NLog.StructuredLogging.Json.Helpers
+{
+    public static class ObjectExtensions
+    {
+        public static bool IsNumeric(this object value)
+        {
+            return value is sbyte
+                   || value is byte
+                   || value is short
+                   || value is ushort
+                   || value is int
+                   || value is uint
+                   || value is long
+                   || value is ulong
+                   || value is float
+                   || value is double
+                   || value is decimal;
+        }
+
```

Comment:
Apologies for the overly verbose method. However, after lots of deliberating this was did seem the best way to prove we were dealing with numerics

Reply:
I would not bother with 2 methods for `IsNumeric` and `IsBoolean` - we don't care about the distinction between bools and numbers here, we just want to know if it is representable as an unquoted value in JSON. 

So one method, call it, e.g. `IsUnquotedValueType` ?